### PR TITLE
Allow filtering with mimetype in Folder::getDirectoryListing

### PR DESCRIPTION
**Pull request by pulsejet**
This is useful e.g. if you need to get all subfolders (exactly what I need)

### CODE REVIEW

**Comment by pulsejet** [comment]
Hmm unfortunately this breaks some apps that inherit from Folder. Maybe a separate method? Or is there a better way to do this?

File: lib/private/Files/Node/Folder.php [hunk 1]
```diff
@@ -93,11 +93,12 @@ public function isSubNode($node) {
 	/**
 	 * get the content of this directory
 	 *
+	 * @param string $mimetype_filter limit returned content to this mimetype or mimepart
```
Comment (CarlSchwan) [inline lib/private/Files/Node/Folder.php:96]:
```suggestion
	 * @param string $mimetypeFilter limit returned content to this mimetype or mimepart
```

Camel case please 🥺

**Comment by kesselb** [comment]
Would you mind to adjust the method signature for `getDirectoryListing` in `lib/private/Files/Node/NonExistingFolder.php`  together with the camelCase change?

Changing the signature is possible for Nextcloud 26. We try to keep our api as stable as possible hence we need a good reason for an signature change.

File: lib/private/Files/Node/Folder.php (hunk 1 above)
Comment (pulsejet) [reply lib/private/Files/Node/Folder.php]: Done. Just want to point out that I took this from the upstream call https://github.com/nextcloud/server/blob/604c1752845df068a7dd5d168abfbfc04065ac3f/lib/private/Files/View.php#L1428

**Comment by pulsejet** [comment]
@kesselb done.

For a "good reason" I should think this qualifies? I need to get just the subfolders of a directory efficiently, and didn't find any other way to do that (using reflection right now).

**Pull request closed without merging**